pkg/operator/config.go

```diff
@@ -82,6 +91,22 @@ func getProviderControllerFromImages(platform configv1.PlatformType, images Imag
 	}
 }
 
+// When the platform type is BareMetal, MAO is also responsible for deploying
+// the baremetal-operator pod
+func getAdditionalProviderImages(platform configv1.PlatformType, images Images) (bool, []string, error) {
+        noImages := []string{"", ""}
+	switch platform {
```

Comment:
We shouldn't be checking the platform here.

Reply:
Why is this resolved? comments does not seems to be addressed. This is called `getAdditionalProviderImages` but this only affect baremetal, naming should reflect that and this can be called only if relevant at the time of building `maoConfigFromInfrastructure`

Reply:
The way I look at this is that operator.go just concerns itself with bringing up the different components of the pod. It really does not concern itself with the underlying platform type. (For example it doesn't care which platform specific CAP* is running.)
Config.go on the other hand is aware of the platform type and I am trying to maintain the same abstraction. It makes sense to me that config.go is still the component that makes decisions based on platform type including whether additional baremetal images are required.
That is also the reason I gave the function a generic sounding name because it will actually be called for all platform type. It will be a no-op for all platform types except baremetal.

If you and @michaelgugino still see it differently, I am happy to make the change to reflect the other way to implement this.

Reply:
I see! although you are well trying to hide the details in the function, skew is already introduced as you need to set `BaremetalPodConfig` and although you gave a neutral name to the function you are defining `baremetalImages` as the variable name. This `AdditionalProviderImages` is only baremetal specific today so I'd keep it simple and explicit rather than over abstract earlier hence my suggestion to do it that way. "Tomorrow" if this happens to be a common patter across providers we can try to go smarter, come up with a proper abstraction and have a generic `AdditionalProviderImages` cross platform